core/base/inc/TBuffer.h

```diff
@@ -76,27 +80,53 @@ class TBuffer : public TObject {
 
    TBuffer(EMode mode);
    TBuffer(EMode mode, Int_t bufsiz);
-   TBuffer(EMode mode, Int_t bufsiz, void *buf, Bool_t adopt = kTRUE, ReAllocCharFun_t reallocfunc = 0);
+   TBuffer(EMode mode, Int_t bufsiz, void *buf, Bool_t adopt = kTRUE, ReAllocStateFun_t reallocfunc = 0, void *reallocData = nullptr);
    virtual ~TBuffer();
 
    Int_t    GetBufferVersion() const { return fVersion; }
    Bool_t   IsReading() const { return (fMode & kWrite) == 0; }
    Bool_t   IsWriting() const { return (fMode & kWrite) != 0; }
    void     SetReadMode();
    void     SetWriteMode();
-   void     SetBuffer(void *buf, UInt_t bufsiz = 0, Bool_t adopt = kTRUE, ReAllocCharFun_t reallocfunc = 0);
-   ReAllocCharFun_t GetReAllocFunc() const;
-   void     SetReAllocFunc(ReAllocCharFun_t reallocfunc = 0);
+   void     SetBuffer(void *buf, UInt_t bufsiz = 0, Bool_t adopt = kTRUE, ReAllocStateFun_t reallocfunc = 0, void *reallocData = nullptr);
+   ReAllocStateFun_t GetReAllocFunc() const;
+   void    *GetReAllocData() const;
+   void     SetReAllocFunc(ReAllocStateFun_t reallocfunc = 0, void *reallocData = nullptr);
    void     SetBufferOffset(Int_t offset = 0) { fBufCur = fBuffer+offset; }
    void     SetParent(TObject *parent);
    TObject *GetParent()  const;
    char    *Buffer()     const { return fBuffer; }
+   char    *GetCurrent() const { return fBufCur; }
    Int_t    BufferSize() const { return fBufSize; }
    void     DetachBuffer() { fBuffer = 0; }
    Int_t    Length()     const { return (Int_t)(fBufCur - fBuffer); }
    void     Expand(Int_t newsize, Bool_t copy = kTRUE);  // expand buffer to newsize
    void     AutoExpand(Int_t size_needed);  // expand buffer to newsize
 
+////////////////////////////////////////////////////////////////////////////////
+/// Share the underlying memory allocation with another buffer.
+//
+// This causes the passed TBuffer object to share our memory buffer.  This is
+// useful if two objects want to have their own view of the TBuffer state but
+// see identical data.
+//
+// Internally, not only do both buffers get the same memory location but a
+// resize done by the "slave" buffer updates the "owner" buffer (the opposite
+// is not true!).
+//
+// This is most useful if the "slave" does all the writings and the "owner"
+// only does reads.
+//
+   virtual void       SetSlaveBuffer(TBuffer &other) { // Share the underlying memory allocation with another TBuffer object.
+       if (other.Buffer() == Buffer()) {return;}
+       Int_t other_buf_size = BufferSize();
+       other.Expand(0, false); // Free up other buffer.
+       other.fBuffer = fBuffer;
+       ResetBit(other.kIsOwner);
+       other.SetReAllocFunc(R__ReAllocShared, this);
+       if (other_buf_size > BufferSize()) {other.Expand(other_buf_size);}
```

Comment:
this is never triggered and if it did would do the 'wrong' things (actually do a reallocation).